### ISSUE TITLE
Sanitize database...

### ISSUE THREAD

**Issue by robbiethegeek** [NYC-Camp/website#34]
We should be sanitizing the DB when pulled from prod to dev so they pullsite will have a sanitized DB, there are lots of ways to do it, but we should totally not have real email addresses, passwords and we should reset user 1 to a known username, email and password for dev / locals that is not what is on prod.

**Comment by esod** [comment]
I totally agree with this, but others thought differently.

**Comment by ForestMars** [comment]
> "there are lots of ways to do it,"

IMO "drush --sanitize" is probably the most  straight-forward / easiest.

**Comment by robbiethegeek** [comment]
drush sql-sync isn't supported on Pantheon, but there is lots of ways to run the SQL that is in --santize to resolve, and drush pullsite doesn't support the --sanitize flag.  Maybe we look at porting the --sanitize flag to drush pullsite command? or adding a callback function to drush pullsite that can run a shell script.

**Comment by ForestMars** [comment]
@Robbie- Actually, the patch for that was written by NYC Community member Chris Oden. I tested it almost 2 years ago (and have been using it since.) It's called Pandarus, and it's super easy to install. 

There are a couple steps, but basically the panderus file to your drush folder allows you to use "drush  pan-sql-sync" against Pantheon as you would normally drush sql-sync, without jumping through any additional hoops.

**Comment by robbiethegeek** [comment]
Let's not keep these beautiful successes under a bushel and get them submitted to the tools, so they can be used by the community :) Please close when you submit or convince Chris Oden too, and document the change since all current developers are pulling DBs full of personally identifiable info to their locals.

**Comment by rabellamy** [comment]
https://github.com/NYC-Camp/website/blob/master/sites/all/drush/pandarus_backend.drush.inc

**Comment by ForestMars** [comment]
Ok, so what do we need to do to incorporate Pandarus into the current workflow?

It seems like we would just need to rewrite scripts to use pan-sql-sync, and update the workflow document. What else?

**Comment by robbiethegeek** [comment]
Am I missing something? Does this support a flag for sanitize?  I just dug through the code and don't see it.

**Comment by ElijahLynn** [comment]
https://drupal.org/project/Pandarus

**Comment by ElijahLynn** [comment]
Also:

http://helpdesk.getpantheon.com/customer/portal/articles/411388-using-drush-on-pantheon

At this time, you cannot use sql-sync on Pantheon. Instead, Pantheon has enabled sql-sync-pipe, a functionally equivalent command. You will need to install Drush SQL Sync Pipe locally in order to use it.

If you need support for sql-sync-pipe, please visit the Drush SQL Sync Pipe issue queue and create a new issue if necessary.

**Comment by ForestMars** [comment]
@likewhoa confirmed he has this (pan-sql-sync) currently working on his local, however we may want to use a pipe instead. cf. https://drupal.org/project/drush_sql_sync_pipe

While Pandarus is a workable solution, it seems like using sql-pipe might be the better way to go here. 

Having said that, we should choose the solution that is _better_ for our workflow (least friction) and not on the basis of ideal / "best practice."

**Also, Elijah just mentioned there may be a speed problem with sql-pipe, which would be a deal-breaker: https://drupal.org/node/2050523

**Comment by robbiethegeek** [comment]
I don't care what we do, what I do care about is obscuring user data.  Right now any dev that signs on to help gets every users email address and password from everyone that attended the last NYC Camp.  That seems like we are not taking seriously the security of people's user information.  This started cause I pointed out a problem, and mentioned there are lots of ways to solve it.  All I care about is that when we do a drush pullsite *dev that we get a database with obfuscated user info, and maybe we reset the admin user to a specific non-production password (and possibly email).

**Comment by likewhoa** [comment]
@robbiethegeek sounds good but what stops users who _we_ grant access to pantheon from fetching the db themselves? +1 for sanitizing the db though with 'drush --sanitize'

I am going to start using sql-pipe tomorrow as I never used it before.

**Comment by ForestMars** [comment]
@likewhoa Let us know what the performance benchmarks for sql-pipe look like. That seems like it's make-or-break.

**Comment by likewhoa** [comment]
@ForestMars tested sql-sync-pipe and the data got corrupted on the first run see  https://gist.github.com/anonymous/9197937 so I ran the command a second time and all when well. 

The speed on sql-sync-pipe is hella slow, see my timings below.

-> time drush -y pan-sql-sync @pantheon.nyccamp2014.live @self
real    1m49.074s
user    0m5.011s
sys 0m2.328s

-> time drush -y sql-sync-pipe @pantheon.nyccamp2014.live @self --progress
19.3MiB 0:05:42 [57.7kiB/s] [                                         <=>                                                      ]

real    5m53.525s
user    0m10.766s
sys 0m6.466s

In conclusion I am putting my recommendation for using <code>drush -y pan-sql-sync --sanitize @pantheon.nyccamp2014.live @self</code>. I will go add instructions to the git workflow documents on how to setup both utilities but I recommend pan-sql-sync.

in terms of forcing a --sanitize option it won't be possible but I can setup a bash wrapper that will add --sanitize for any drush command that includes _sync_.

We just have to tell users to use this wrapper or they get no cookies!

**Comment by ForestMars** [comment]
@likewhoa @ElijahLynn So we have definitely confirmed that pan-sql-sync supports the --sanitize flag…?

**Comment by likewhoa** [comment]
confirmed that --sanitize works with pan-sql-sync. I don't recommend sql-sync-pipe at this time.

**Comment by mirie** [comment]
@ElijahLynn, @ForestMars - Confirmed with @likewhoa help that pan-sql-sync --sanitize works. We will update that tools document today. 

@ElijahLynn, is that document ready to be added to the readme after we add this part?

**Comment by mirie** [comment]
@jmarkel also confirmed that this worked for kalastack users with no additional configuration necessary.

**Comment by mirie** [comment]
All, the writeup is complete in the Git Workflow & Dev Environment document. Not sure what our preference is for where this document should live, so leaving this issue open until that final piece is decided.

**Comment by ForestMars** [comment]
I think the plan was to make it a (wiki) page on Github (with Github pages, ie, not a repository.) 

I recall someone volunteering to do that, who was it!

**Comment by ElijahLynn** [comment]
I volunteered, it likely won't be until Thursday night before I can convert
it. The main workflow steps should go in the readme.MD if anyone wants to
do it before then.

Elijah Lynn
903.Eli.jahs (354.5247)
www.elijahlynn.net

**Comment by jmarkel** [comment]
The first pass at markdown-ing the overview is in PR #42

**Comment by mirie** [comment]
@jmarkel, can up date the readme so that feature branches are made off of master, and not integration?

**Comment by jmarkel** [comment]
@mirie Sure, but deferring to the weekend (see @jonpugh 's comment on #61)

**Comment by mirie** [comment]
ALL

Please review the updated wiki: https://github.com/NYC-Camp/website/wiki

**Comment by likewhoa** [comment]
@mirie :+1:

**Comment by mirie** [comment]
Closing.
